### ISSUE TITLE
Switch to (stateless) functional components

### ISSUE THREAD

**Issue by MoOx** [MoOx/phenomic#46]
...instead of classes.

**Comment by MoOx** [comment]
Not a good idea yet since hot loading is kind of broken with stateless component

**Comment by MoOx** [comment]
Blocked by https://github.com/gaearon/babel-plugin-react-transform/issues/57

**Comment by thangngoc89** [comment]
https://github.com/gaearon/babel-plugin-react-transform/pull/85

**Comment by MoOx** [comment]
The solution is pretty filthy and not reliable imo. Not even sure this will be merged.

This comment intrigued mehttps://github.com/gaearon/babel-plugin-react-transform/pull/85#issuecomment-185048429

This makes me start thinking that HMR is not optimal.
See this thread https://twitter.com/thejameskyle/status/699844680628834304

I will look at http://chromakode.github.io/isolated-core/.

**Comment by thangngoc89** [comment]
Interesting. 
Keep this here just for the record https://github.com/thejameskyle/isolated-core-demo

**Comment by MoOx** [comment]
FYI https://github.com/gaearon/babel-plugin-react-transform/issues/57#issuecomment-190038433

**Comment by thangngoc89** [comment]
Awesome. Let's move everything to redux store.

**Comment by MoOx** [comment]
Still unsure that we need redux. Most things don't even changes (collections, layouts etc). Only pages data, and redux still not really relevant since we don't do complexe changes, but just add/replace in an object...

**Comment by thangngoc89** [comment]
Sorry. I didn't review that PR carefully. So @-gaearon is re-rendering the whole app on each hot reload right ?

**Comment by MoOx** [comment]
Yeah. But I think if we try this, we should consider http://chromakode.github.io/isolated-core/

**Comment by thangngoc89** [comment]
Isolated core mount app to an iframe while ReactDOM mount app to DOM. Not sure what's the best. Anyway, will give a try tmr.

**Comment by MoOx** [comment]
Meanwhile, nice light example https://github.com/Winwardo/solid-octo-disco/commit/e2f22b2c915e6542dd819cfeac21e43ad261048c

**Comment by thangngoc89** [comment]
He killed his project again https://medium.com/@dan_abramov/hot-reloading-in-react-1140438583bf#.77dfvm6qh

**Comment by gaearon** [comment]
React Hot Loader 3 supports functional components without destroying the state.
It supersedes both React Hot Loader and React Transform, and is built on lessons learned from both.
Check it out: https://github.com/gaearon/react-hot-boilerplate/pull/61

**Comment by thangngoc89** [comment]
@gaearon :+1: :+1: best news ever

**Comment by MoOx** [comment]
webpack-2 branch now use react-hot-loader@3, so we are now able to use stateless component everywhere when possible!

**Comment by MoOx** [comment]
webpack 2 and react-hot-loader v3 are on master, so contributions are welcome:
- remove this rule https://github.com/MoOx/phenomic/blob/eb31532bc3d3dd20ec5238754b0868476df431bc/package.json#L233-L236
- remove this rule in docs & default theme template
- update components!

**Comment by MoOx** [comment]
Closing in favor of #747
